### PR TITLE
[ios] 14539 Type Based API for Date 

### PR DESCRIPTION
**Pull request by adam-govan**
As part of this story, I had to add the ability to pass a NSDate as the User Update event. 

there is a new function: 
and for  one: 
`- (int) userUpdate:(NSString *) name withDate:(NSDate *) date;`

@Sergio-Mira / @dominicmarmionswrve  / @seaders

### CODE REVIEW

File: SwrveSDK/SDK/Track/Swrve.h [hunk 1]
```diff
@@ -651,6 +651,17 @@ typedef void (^SwrveResourcesUpdatedListener) ();
  */
 -(int) userUpdate:(NSDictionary*)attributes;
 
+/*! Sends a single Date based user state to Swrve
+ *
+ * See the REST API docs for the user event for a detailed description of the
+ * semantics of this call.
+ *
+ * \param date The NSDate value associated
+ * \param name The identifier for the user update
```
Comment (Sergio-Mira) [inline SwrveSDK/SDK/Track/Swrve.h:659]:
~~~ param name first?

**Comment by Sergio-Mira** [comment]
🚀

File: SwrveSDK/SDK/Track/Swrve.h (hunk 1 above)
Comment (adam-govan) [reply SwrveSDK/SDK/Track/Swrve.h:659]: good one! i'll fix that now

File: SwrveSDK/SDK/Track/Swrve.h [hunk 2]
```diff
@@ -651,6 +651,17 @@ typedef void (^SwrveResourcesUpdatedListener) ();
...
+/*! Sends a single Date based user state to Swrve
+ *
+ * See the REST API docs for the user event for a detailed description of the
+ * semantics of this call.
+ *
+ * \param name The identifier for the user update
+ * \param date The NSDate value associated
+ * \returns SWRVE_SUCCESS if the call was successful, otherwise SWRVE_ERROR.
+ */
+- (int) userUpdate:(NSString *)name withDate:(NSDate *) date;
```
Comment (dominicmarmionswrve) [inline SwrveSDK/SDK/Track/Swrve.h:663]: 👍

File: SwrveSDK/SDK/Track/Swrve.m [hunk 3]
```diff
@@ -1094,6 +1094,30 @@ -(int) userUpdate:(NSDictionary*)attributes
...
 
+- (int) userUpdate:(NSString *)name withDate:(NSDate *) date {
+    
+    if(name && date){
+        NSMutableDictionary * currentAttributes = (NSMutableDictionary*)[self.userUpdates objectForKey:@"attributes"];
+        [self.userUpdates setValue:[NSNumber numberWithUnsignedLongLong:[self getTime]] forKey:@"time"];
+        [currentAttributes setObject:[self convertDateToString:date] forKey:name];
+        
+    }else{
+        DebugLog(@"nil object passed into userUpdate:withDate");
```
Comment (dominicmarmionswrve) [inline SwrveSDK/SDK/Track/Swrve.m:1105]: 👍

**Comment by dominicmarmionswrve** [comment]
🚀

**Pull request merged**